wp_derive_request_builder/src/generate.rs

```diff
@@ -78,7 +80,17 @@ fn generate_async_request_executor(
                 pub async #fn_signature -> Result<#response_type_ident, #error_type> {
                     use #crate_ident::middleware::PerformsRequests;
                     #request_from_request_builder
-                    self.perform(std::sync::Arc::new(request)).await?.parse()
+                    let response = self.perform(std::sync::Arc::new(request)).await?;
+                    let response_status_code = response.status_code;
+                    let parsed_result: Result<#response_type_ident, #error_type> = response.parse();
+                    if let Err(ref err) = parsed_result {
+                        if let Some(wp_error_code) = err.wp_error_code() {
+                            if #crate_ident::is_unauthorized_request(response_status_code, wp_error_code, self.request_executor.clone(), self.api_root_url.clone(), self.authentication.clone()).await {
```

Comment:
As discussed above, these if statements can be changed to `if is_unauthorized_request == 401 && #crate_ident::is_unauthorized_request(...) {`